### ISSUE TITLE
False positive CA2208 in extension block

### ISSUE THREAD

**Issue by andryushchenko** [dotnet/roslyn-analyzers#7773]
### Analyzer
Diagnostic ID: [CA2208](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2208): Instantiate argument exceptions correctly

### Analyzer source
**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

Version: 10.0.100

### Describe the bug

False positive CA2208 for extension block argument if extension method has other arguments

```
public static class ExtensionTest
{
    extension(object item)
    {
        public void Test1()
        {
            _ = item ?? throw new ArgumentNullException(nameof(item)); // OK
        }
        
        public void Test2(object argument)
        {
            _ = item ?? throw new ArgumentNullException(nameof(item)); // CA2208
            _ = argument ?? throw new ArgumentNullException(nameof(argument)); // OK
        }
    }
}
```

### Expected behavior
No warnings

### Actual behavior
False positive
